### PR TITLE
Implemented skip reason to show in categories and temporary solution for "TypeError: allureJsCommons.AllureConfig is not a constructor" error to lock on "allure-js-commons" version that does not have an error

### PR DESCRIPTION
**Pull request by lpelypenko**
Current Pull request had two changes: 

**Change 1 is a temporary bug fix for error "TypeError: allureJsCommons.AllureConfig is not a constructor"** 
We started to get an error "TypeError: allureJsCommons.AllureConfig is not a constructor" that fails our builds and makes library unusable for us.

Until the issue is fixed in https://github.com/allure-framework/allure-js-commons/issues/48 I would like to propose this temporary solution to lock on  "allure-js-commons": "2.0.0-beta.8" that is still backward compatible.

Removing `^` should lock npm on specific version, while using "^2.0.0-beta.8" currently intstalls minor versions that incompatible with current implementation of `testcafe-reporter-allure`

The error is also reported in https://github.com/isaaceindhoven/testcafe-reporter-allure/issues/102. 

**Change 2 is a new feature to allow skipReason metadata that will improve Categories view and skipped test overiview**

TestCafe does not support skip message in it's API and skipped tests does not have any way to show skipped message in allure report. 

In addition, if you want to filter skipped tests by skip reasons that could be `known product bug`, `test bug` or `not automated test`, it is not possible because allure allows to filter only by `matchedStatuses`, `messageRegex` and `traceRegex`. Without skip message/reason we can't filter them in different categories. (see documentation https://docs.qameta.io/allure/#_categories)

But somehow python allure integration writes skip reason that is looking like this in raw code:

```json
  "status": "skipped",
  "statusDetails": {
    "message": "Skipped: Skipped by decorator",
    "trace": "('/pytest/tests/test_example.py', 51, 'Skipped: Skipped by decorator')"
  },
```
I assumed that allure will not see it, but when report was generated it showed skip reason nicely in categories and I implemented it in testCafe also:

Screen Shot 2021-09-02 at 12 00 22

Summary of changes:

- added support of custom meta skipReason
- if skipReason is provided, message will be set to "Skipped: ${skip reason}",
```
  "status": "skipped",
  "statusDetails": { "message": "Skipped: Not automated", "trace": "" },
```
- added set of examples with custom report and skip reason
- updated documentation

Summary of new feature:
 
If use skip the test and add `skipReason` meta, value of `skipReason` will be added to raw allure report message. Example allure raw result: 

```
{
  "uuid": "6f3538b3-faa7-4e30-aa39-1c11152fccf3",
  "historyId": "4443866b-4eea-45ef-bf8b-2ecdf6571adf",
  "status": "skipped",
  "statusDetails": { "message": "Skipped: Not automated", "trace": "" },
  "stage": "finished",
  "steps": [],
  "attachments": [],
  "parameters": [{ "name": "skipReason", "value": "Skipped: Not automated" }],
  "labels": [
    { "name": "framework", "value": "TestCafe" },
    { "name": "language", "value": "typescript/javascript" },
    { "name": "severity", "value": "trivial" },
    { "name": "suite", "value": "TestCafé Example Fixture - Documentation" },
    { "name": "subSuite", "value": "Main Example Group" },
    { "name": "epic", "value": "Example Epic Ticket" },
    { "name": "feature", "value": "Example Feature Ticket" },
    { "name": "story", "value": "Example Story Ticket" }
  ],
  "links": [{ "name": "JIRA Issue: TEST-ISSUE", "url": "https://jira.example.nl/browse/TEST-ISSUE", "type": "issue" }],
  "start": 1630560603492,
  "name": "Example planned test",
  "fullName": "TestCafé Example Fixture - Documentation : Example planned test",
  "description": "An example description",
  "stop": 1630560603492
}

```

What do you think about both changes?

### CODE REVIEW

**Review by vanhoofmaarten — CHANGES_REQUESTED** [review]
Hi @lpelypenko,

Thank you very much for the thorough description and your time to submit this PR.

Some feedback:
1. Could you split up the changes into separate branches?
	1. "TypeError: allureJsCommons.AllureConfig is not a constructor" error
	2. Implemented skip reason to show in categories
1. I've noticed a few linting errors are still present in the provided code. This is an error on our side. The safeguards (lint-staged / GitHub Action) system does a lint and autofix, without committing the changed auto-fixed files. This is now fixed in the develop branch. Could you merge the updated develop branch in this one and run 'npm install'?

Moving the `allureJsCommons.AllureConfig is not a constructor` fix seems to be a relatively quick change?

**Comment by lpelypenko** [comment]
Hi @vanhoofmaarten,

I really appreciate your fast reply. 
Thank you for suggestion to split changes in 2 separate commits. This makes more sense to me too. I also had it in mind but initially made a mistake pushing all in my `develop` branch. 
I have already pushed temporary fix for type error into this branch https://github.com/isaaceindhoven/testcafe-reporter-allure/pull/108

Will create a separate pull with small feature implementation shortly.

File: docs/usage/metadata.md [hunk 1]
```diff
@@ -27,10 +61,23 @@ test.meta({
...
 
 It is also possible to add custom metadata to a `test`. These will be added as parameters to the Allure Report. These parameters do __NOT__ have to be named otherMeta but do require to be a key-value pair of two strings. For example, adding a color parameter to a test: `color: 'black'` will result in `color: black` be added to the final Allure Report.
 
 ## Result
-![Example of the metadata code shown below.](/images/metadata.PNG)
\ No newline at end of file
+
+## Suite overview
+
+![Example of the metadata for passed test.](../images/passed.png)
```
[the remark this answers is not in the document]
Comment (lpelypenko) [reply docs/usage/metadata.md:74]: Thank you for the suggestion. I have added more details in `docs/usage/metadata.md` and also wrote more explanation on how skip reason can benefit `Categories` view if user used custom allure categories filter. 
Moved changed in https://github.com/isaaceindhoven/testcafe-reporter-allure/pull/109

**Comment by lpelypenko** [comment]
I think this MR should be closed because I split changes in two MRs:
https://github.com/isaaceindhoven/testcafe-reporter-allure/pull/108
https://github.com/isaaceindhoven/testcafe-reporter-allure/pull/109

**Pull request closed without merging**